packages/legacy/core/App/screens/ProofRequest.tsx

```diff
@@ -561,11 +561,17 @@ const ProofRequest: React.FC<ProofRequestProps> = ({ navigation, proofId }) => {
   const handleDeclineTouched = useCallback(async () => {
     try {
       if (agent && proof) {
-        await agent.proofs.sendProblemReport({ proofRecordId: proof.id, description: t('ProofRequest.Declined') })
+        const connectionId = proof.connectionId ?? ''
+        const connection = await agent.connections.findById(connectionId)
+
+        if (connection) {
+          await agent.proofs.sendProblemReport({ proofRecordId: proof.id, description: t('ProofRequest.Declined') })
+        }
+
         await agent.proofs.declineRequest({ proofRecordId: proof.id })
 
-        if (proof.connectionId && goalCode && goalCode.endsWith('verify.once')) {
-          agent.connections.deleteById(proof.connectionId)
+        if (connectionId && goalCode && goalCode.endsWith('verify.once')) {
```

Comment:
@jleach I think another check for whether the connection exists or not is needed here to prevent the error in bcgov/bc-wallet-mobile#2349

Reply:
On like 565 above I fetch the connection and make sure it exists. Good enough?

Reply:
How would you feel about something like this for line 573:

```suggestion
        if (connection && goalCode?.endsWith('verify.once')) {
```

Not too aggressive of a change